src/frontend/src/service/api-service.ts

```diff
@@ -0,0 +1,41 @@
+import axios from 'axios';
+
+type MethodOptions =
+  | 'POST'
+  | 'post'
+  | 'GET'
+  | 'get'
+  | 'DELETE'
+  | 'delete'
+  | 'head'
+  | 'HEAD'
+  | 'put'
+  | 'PUT'
+  | 'patch'
+  | 'PATCH';
+
+type Req = {
+  url: string;
+  data?: object;
+  method: MethodOptions;
+};
+
+export default function apiService(req: Req) {
+  return axios
+    .request(req)
+    .then(res => [res.statusText, res.data, res.status])
+    .then(([statusText, data, status]) => {
+      if (statusText) {
```

Comment:
Given that this is an `if`, when will `statusText` not be truthy? That should be commented.

If we don't expect `statusText` to ever be falsy then IMO we should throw an exception and `alert` if it is.

Reply:
I was under the assumption that if something is to go wrong with the fetch then it wouldn't return statusText, data, and or a status.  I am simply checking if we got a response by checking if statusText is available.  In other words statusText will be truthy as long as the fetch worked properly.  The if statement might as well have checked if data or status is available.  In either case if we don't get a response from the service then the error is caught and returned instead.    Let me know if this makes sense, and or how you would want to modify the code.

Reply:
With TypeScript I had to refactor the code to look like this, else type checks were failing.

 .then((response: Response) => {
      return {
        data: response.data,
        status: response.status,
        statusText: response.statusText
      };
    })